server/bastionlab_polars/src/lib.rs

```diff
@@ -336,6 +346,52 @@ Reason: {}",
         }
         Ok(())
     }
+
+    fn dos_check(
```

Comment:
This could be a method defined on a dedicated Tracking type?

Reply:
I've created a dedicated Tracking type.